DemoApp/DeeplinkAdapter.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  DeeplinkAdapter.swift
+//  DemoApp
+//
+//  Created by Ilias Pavlidakis on 29/5/23.
+//
+
+import Foundation
+import StreamVideo
+
+struct DeeplinkInfo: Equatable {
+    var callId: String
+    var callType: String
+
+    static let empty = DeeplinkInfo(callId: "", callType: "")
+}
+
+struct DeeplinkAdapter {
+    var baseURL: URL
+
+    func canHandle(url: URL) -> Bool {
+        url.absoluteString.contains(baseURL.absoluteString)
+    }
+
+    func handle(url: URL, completion: (DeeplinkInfo, User) -> Void) {
```

Comment:
Why do we need completion here? Can't we just directly return the deeplink info?